docs/user-guide/dream/dream-advanced-powder-reduction.ipynb

```diff
@@ -453,7 +461,134 @@
+    "    if (mask := hist.masks.pop('zero_vanadium', None)) is not None:\n",
+    "        hist.data = sc.where(mask, sc.scalar(0.0), hist.data)\n",
+    "    histograms.append(hist)\n",
+    "\n",
+    "sc.concat(histograms, dim='detector').sum('detector').plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "35",
+   "metadata": {},
+   "source": [
+    "### 2D intensity vs. d-spacing and 2θ\n",
+    "\n",
+    "Just like before, we map the workflow over the detector names.\n",
+    "But here we also assign different $2\\theta$ bins for each detector bank."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "36",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "detector_names = [\"mantle\", \"endcap_forward\", \"endcap_backward\", \"high_resolution\"]\n",
+    "two_theta_bins = [\n",
+    "    sc.linspace(dim=\"two_theta\", unit=\"rad\", start=0.77, stop=2.36, num=201),\n",
+    "    sc.linspace(dim=\"two_theta\", unit=\"rad\", start=0.24, stop=0.71, num=101),\n",
+    "    sc.linspace(dim=\"two_theta\", unit=\"rad\", start=2.42, stop=2.91, num=151),\n",
+    "    sc.linspace(dim=\"two_theta\", unit=\"rad\", start=2.94, stop=3.11, num=51),\n",
+    "]\n",
+    "parameter_table = pd.DataFrame(\n",
+    "    {NeXusDetectorName: detector_names, TwoThetaBins: two_theta_bins},\n",
+    "    index=detector_names\n",
+    ").rename_axis(index='detector')\n",
+    "\n",
+    "all_detector_workflow = workflow.copy()\n",
+    "mapped = all_detector_workflow[IofDspacingTwoTheta].map(parameter_table)\n",
+    "all_detector_workflow[IofDspacingTwoTheta] = mapped.reduce(func=powder.grouping.collect_detectors)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "37",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result = all_detector_workflow.compute(IofDspacingTwoTheta)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "38",
+   "metadata": {},
+   "source": [
+    "We can 'sum' over $2\\theta$ by using `concat` and then sum over all detectors to obtain a 1D curve:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "39",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sum(da.bins.concat('two_theta').hist() for da in result.values()).plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "40",
+   "metadata": {},
+   "source": [
+    "But we can also preserve the $2\\theta$ dimension to produce a 2D plot.\n",
+    "However, the detector banks have different $2\\theta$ ranges.\n",
+    "So we need to merge them into an appropriate array first.\n",
+    "We could concatenate the event lists for all banks directly and then bin into new $2\\theta$ and $d$-spacing bins.\n",
+    "\n",
+    "But we can also preserve the $2\\theta$ binning for each bank to respect differences in resolution.\n",
+    "The following cell does so by filling the gaps between banks with empty event lists.\n",
```

Comment:
It seems very cumbersome to have to manually add empty event lists.
I know you didn't like to wast memory by making common 2theta bins for all, but it wasn't that much memory (the dimensions are not large).
I feel it would be more convenient to work with?

Reply:
It may be. But doing so is also a bit complicated when we want different resolutions for the banks.
Let me check with Céline to see what we actually need.